lttoolbox/compression.cc

```diff
@@ -254,7 +254,7 @@ Compression::multibyte_read(istream &input)
 
 
 void
-Compression::wstring_write(wstring const &str, FILE *output)
+Compression::string_write(UString const &str, FILE *output)
```

Comment:
The changed binary I/O has to be wrong. Old code was UTF-32, new is UTF-16. And this must actually have been fundamentally broken forever - have we never tried loading an FST with alphabets beyond the BMP on Windows?

Reply:
I'm pretty sure it's 32, but the way I did it seems a bit inefficient, but since there aren't converters included between 16 and 32 maybe I'm stuck?

Reply:
`utf8::utf16to32()` and inverse exist. But this must have been broken on Windows before, what with `wchar_t` being UTF-16 there vs. UTF-32 elsewhere. We need tests with code points beyond the BMP to be sure.